server/cluster_worker.go

```diff
@@ -40,71 +40,37 @@ func (c *raftCluster) handleAddPeerReq(region *metapb.Region) (*metapb.Peer, err
 	// check this.
 	// 2, We can check the store statistics and find a low load store.
 	// 3, more algorithms...
-	var matchStore *metapb.Store
+L:
 	for _, store := range mu.stores {
-		storeID := store.GetId()
-
-		existStore := false
+		// we can't add peer in the same store.
 		for _, peer := range region.Peers {
-			if peer.GetStoreId() == storeID {
-				// we can't add peer in the same store.
-				existStore = true
-				break
+			if peer.GetStoreId() == store.GetId() {
+				continue L
 			}
 		}
-
-		if existStore {
-			continue
-		}
-
-		matchStore = &store
-
-		break
-	}
-
-	if matchStore == nil {
-		return nil, errors.Errorf("find no store to add peer for region %v", region)
-	}
-
-	peer := &metapb.Peer{
-		StoreId: proto.Uint64(matchStore.GetId()),
-		Id:      proto.Uint64(peerID),
+		return &metapb.Peer{
+			Id:      proto.Uint64(peerID),
+			StoreId: proto.Uint64(store.GetId()),
+		}, nil
 	}
-
-	region.Peers = append(region.Peers, peer)
-
-	return peer, nil
+	return nil, errors.Errorf("find no store to add peer for region %v", region)
 }
 
 // If leader is nil, we will return an error, or else we can remove none leader peer.
-func (c *raftCluster) handleRemovePeerReq(region *metapb.Region, leader *metapb.Peer) (*metapb.Peer, error) {
+func (c *raftCluster) handleRemovePeerReq(region *metapb.Region, leaderID uint64) (*metapb.Peer, error) {
```

Comment:
@qiuyesuifeng I think it's acceptable not to check it. And if we want to, maybe better to check it outside of this func.